src/Foundation/Enums.cs

```diff
@@ -1,7 +1,13 @@
 using XamCore.ObjCRuntime;
+using XamCore.CloudKit;
 
 namespace XamCore.Foundation {
 
+#if MONOMAC
+	delegate void CloudKitRegistrationPreparationAction (CloudKitRegistrationPreparationHandler handler);
```

Comment:
I think you are missing `[BlockCallback]` here

`delegate void CloudKitRegistrationPreparationAction ([BlockCallback] CloudKitRegistrationPreparationHandler handler);` 

Since the definition is

`- (void)registerCloudKitShareWithPreparationHandler:(void (^)(void (^preparationCompletionHandler)(CKShare *, CKContainer *, NSError *)))preparationHandler;`

You will get a Block instead of a CCalback and IIRC we default to CCalbacks if the attribute is not present. ([I hate the block syntax](http://fuckingblocksyntax.com/)).